src/test/java/io/jenkins/plugins/eddsa_api/EddsaTest.java

```diff
@@ -37,26 +39,32 @@ public class EddsaTest implements Serializable {
     public static final String USER = "jenkins";
 
     @Rule
-    public transient GenericContainer agentContainer = new GenericContainer(new ImageFromDockerfile(
-                            SSH_AGENT_NAME, false)
-                    .withFileFromClasspath(
-                            SSH_AUTHORIZED_KEYS,
-                            AGENTS_RESOURCES_PATH + "/" + SSH_AGENT_NAME + "/" + SSH_AUTHORIZED_KEYS)
-                    .withFileFromClasspath(
-                            SSH_KEY_PATH, AGENTS_RESOURCES_PATH + "/" + SSH_AGENT_NAME + "/" + SSH_KEY_PATH)
-                    .withFileFromClasspath(
-                            SSH_KEY_PUB_PATH, AGENTS_RESOURCES_PATH + "/" + SSH_AGENT_NAME + "/" + SSH_KEY_PUB_PATH)
-                    .withFileFromClasspath(
-                            SSH_SSHD_CONFIG, AGENTS_RESOURCES_PATH + "/" + SSH_AGENT_NAME + "/" + SSH_SSHD_CONFIG)
-                    .withFileFromClasspath(DOCKERFILE, AGENTS_RESOURCES_PATH + "/" + SSH_AGENT_NAME + "/" + DOCKERFILE))
-            .withExposedPorts(22);
+    public transient GenericContainer agentContainer = !DockerClientFactory.instance()
+                    .isDockerAvailable()
+            ? null
+            : new GenericContainer(new ImageFromDockerfile(SSH_AGENT_NAME, false)
+                            .withFileFromClasspath(
+                                    SSH_AUTHORIZED_KEYS,
+                                    AGENTS_RESOURCES_PATH + "/" + SSH_AGENT_NAME + "/" + SSH_AUTHORIZED_KEYS)
+                            .withFileFromClasspath(
+                                    SSH_KEY_PATH, AGENTS_RESOURCES_PATH + "/" + SSH_AGENT_NAME + "/" + SSH_KEY_PATH)
+                            .withFileFromClasspath(
+                                    SSH_KEY_PUB_PATH,
+                                    AGENTS_RESOURCES_PATH + "/" + SSH_AGENT_NAME + "/" + SSH_KEY_PUB_PATH)
+                            .withFileFromClasspath(
+                                    SSH_SSHD_CONFIG,
+                                    AGENTS_RESOURCES_PATH + "/" + SSH_AGENT_NAME + "/" + SSH_SSHD_CONFIG)
+                            .withFileFromClasspath(
+                                    DOCKERFILE, AGENTS_RESOURCES_PATH + "/" + SSH_AGENT_NAME + "/" + DOCKERFILE))
+                    .withExposedPorts(22);
 
     @Rule(order = 10)
     public transient RealJenkinsRule j =
             new RealJenkinsRule().withDebugPort(8000).withDebugServer(true).withDebugSuspend(true);
 
     @Test
     public void connectionTests() throws Throwable {
+        assumeTrue(DockerClientFactory.instance().isDockerAvailable());
```

Comment:
NIT: it is always good to know /why/ a test is skipped... 
`assumeTrue("No docker instance was not found or available", DockerClientFactory.instance().isDockerAvailable());`